sound/soc/sof/loader.c

```diff
@@ -181,6 +193,9 @@ static int snd_sof_fw_ext_man_parse(struct snd_sof_dev *sdev,
 
 		/* process structure data */
 		switch (elem_hdr->type) {
+		case EXT_MAN_ELEM_FW_VERSION:
+			ret = ext_man_get_fw_version(sdev, elem_hdr);
+			break;
```

Comment:
@ktrzcinx This is a bit iffy to write the version to "fw_ready", but I guess it's ok as we don't have anything else to do it with now. Maybe printing it out would be useful (or is it done elsewhere already now)?

Reply:
Yes, it is quite iffy but it's how it was done before  - `fw_ready` contains `version` - I wouldn't like to break backward compatibility. Printing it out is done in `snd_sof_ipc_valid()` function, also iffy, also kept as before because it's not a target for this PR.

Reply:
Ack @ktrzcinx I think this is ok. The snd_sof_ipc_valid() function name is a bit misleading as well, but that's not something added in this patch.